### DISCUSSION TITLE
2022-04-15 Developer Tools Update

### DISCUSSION THREAD

**Discussion by austinbaggio** [near/community#250] [Developer Tools]
The developer tools team focuses on NEAR's tools, APIs, & SDKs for both Rust and Javascript. 

## Tools Feedback 
You can reach out to Austin Baggio (Tools Product Manager) to share any feedback about this update or your experience using any of our Dev Tools:  
[30m Calendly Timeslot](https://calendly.com/austinbaggio/30min) | austin.baggio@near.org | [Discord](https://discord.com/channels/490367152054992913/542945453533036544) | [Twitter](https://twitter.com/AustinBaggio)

Join us this Thursday for a Dev Tools Deep Dive! Members of the Developer Tools team will take an hour to discuss what we're working on, our plans for the future, and most importantly - answer questions from the community! More details to be posted Monday in the [NEAR Discord](https://discord.com/channels/490367152054992913/542945453533036544)

### workspaces-rs
- @itegulov is developing a near-sdk-sim -> workspaces-rs migration document. Workspaces RS & JS are going to be the long term supported testing frameworks. 
- @ChaoticTempest Added betanet support to workspaces-rs: https://github.com/near/workspaces-rs/pull/116 Custom networks are to be added in the coming weeks 

### workspaces-js
- @ailisp Modified the account creation API and getAccount API to take the full account name without implicit suffix. Added createSubAccount and getSubAccount to explicitly add subaccounts
- @ailisp Fixed the CI definition to run testnet tests
@ailisp Fixed some cases of the "Can not sign transactions ... no matching key pair found" issue which appeared in most CI test and user reported test on testnet. There are still remaining issues that result in this error to be investigated in the coming weeks
- @volovyk-s Cleaned up the workspaces-js interface https://github.com/near/workspaces-js/pull/137 & https://github.com/near/workspaces-js/pull/136
- @volovyk-s Investigated and added the ability to track the non-precise location of NEAR CLI users in Mixpannel:
https://github.com/near/near-cli/pull/962

### jsonrpc-client
- @miraclx Fixed query method deserialization https://github.com/near/near-jsonrpc-client-rs/pull/82
- @miraclx Reviewed more jsonrpc-client documentation PRs, and continued investigating jsonrpc fuzzing and why tokio's reactor is failing to spawn inside a fuzzing thread

### nearcore
- @miraclx Updated actix on nearcore https://github.com/near/nearcore/pull/6573 and is updating actix-web hoping to fix the JSON RPC fuzzing issue
- @austinabell e2e tested the nep264 change and stabilized the API for next release https://github.com/near/nearcore/pull/6564

### near-sdk-rs
- @itegulov Added support for arm64 in Rust SDK example build scripts: https://github.com/near/near-sdk-rs/pull/758[](https://github.com/near/near-sdk-rs/pull/758)
Simplified some logic in ACI proof of concept and made sure the results are reproducible in contract-builder image
- @austinabell Is updating workspaces to 0.2 in near-sdk-rs. Currently working through a debugging inconsistency with @ChaoticTempest https://github.com/near/near-sdk-rs/pull/789
- @ChaoticTempest is investigating weird behavior with Austin on the newest nearcore binaries causing cross-contract calls to fail, specifically factor contract in near-sdk-rs
- @ChaoticTempest Released workspaces-rs 0.2.1 to fix [docs.rs](http://docs.rs/) builds from failing: https://github.com/near/workspaces-rs/pull/117

### ACI 
- @itegulov Simplified some logic in ACI proof of concept and made sure the results are reproducible in contract-builder image

**Comment by maxzaver** [comment]
@AustinBaggio should we name our metadata/ACI as ABI to be consistent with Ethereum naming convention and make it more discoverable? @thisisjoshford @dgorton

**Reply by austinbaggio** [reply]
Currently an open discussion. Do you have insight into the original decision behind ACI & metadata as the current naming convention?

**Reply by itegulov** [reply]
To me ABI implies something very low-level (which Ethereum needs because their virtual machine has a notion of elementary types, function selectors etc), but we are building on top of WASM that takes care of all that. At the same time ACI is indeed a term with low discoverability. From what I have seen other L1s refer to this as simply metadata (e.g. Polkadot, Solana), which is not very descriptive either but we might want to follow this ad-hoc convention.

**Comment by maxzaver** [comment]
> @ChaoticTempest is investigating weird behavior with Austin on the newest nearcore binaries causing cross-contract calls to fail, specifically factor contract in near-sdk-rs

Are we talking about Rust SDK code that community is already using for smart contract development? @AustinBaggio If yes, do we have a plan for notifying the community?

**Reply by austinabell** [reply]
no, has nothing to do with the SDK

**Reply by ChaoticTempest** [reply]
yeah this doesn't have anything to do with the SDK itself, but just some weird behaviors with creating accounts through contract promises that don't meet the top level account naming rules such as too short of a length. The old sandbox binaries (dating back since october of last year) worked for short names for some reason while the newest ones fixes this issue
